compiler/noirc_evaluator/ssa_test/src/main.rs

```diff
@@ -0,0 +1,111 @@
+use std::fs::File;
+use std::io::Write;
+use std::io::Read;
+use std::path::Path;
+use flate2::read::GzDecoder;
+use::noirc_evaluator::acir_instruction_builder::{
+    InstructionArtifacts, VariableType, Variable
+};
+use clap::Parser;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long, default_value = "../../../../../barretenberg/cpp/src/barretenberg/acir_formal_proofs/artifacts/")]
+    dir: String,
+}
+
+/// Decompresses gzipped data into a byte vector
+fn ungzip(compressed_data: Vec<u8>) -> Vec<u8> {
+    let mut decompressed_data: Vec<u8> = Vec::new();
+    let mut decoder = GzDecoder::new(&compressed_data[..]);
```

Comment:
I looks like `unzip` could take a reference, and then the `acir.clone()` in `save_artifacts` could be just `&acir`.